src/flux/model.py

```diff
@@ -3,14 +3,31 @@
 import torch
 from torch import Tensor, nn
 
-from flux.modules.layers import (DoubleStreamBlock, EmbedND, LastLayer,
-                                 MLPEmbedder, SingleStreamBlock,
-                                 timestep_embedding)
-
+from flux.modules.layers import (
+    EmbedND,
+    LastLayer,
+    MLPEmbedder,
+    timestep_embedding,
+)
+from flux.modules.lora import LinearLora, replace_linear_with_lora
+
+try:
+    import triton_kernels
+    from triton_kernels import SingleStreamBlock, DoubleStreamBlock
+except ImportError:
+    print("Triton kernels not found, using flux native implementation.")
+    from flux.modules.layers import SingleStreamBlock, DoubleStreamBlock
+except ModuleNotFoundError:
+    print("Triton kernels not found, using flux native implementation.")
+    from flux.modules.layers import SingleStreamBlock, DoubleStreamBlock
+except Exception as e:
+    print(f"Error: {e}")
+    from flux.modules.layers import SingleStreamBlock, DoubleStreamBlock
```

Comment:
Added graceful fallback mechanism for triton kernel imports - if triton kernels are unavailable or fail to load, the code automatically falls back to native FLUX implementation while providing appropriate error messages. This ensures smoother execution across different environments and clearer debugging information.